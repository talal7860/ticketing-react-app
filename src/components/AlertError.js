import React from 'react';
import { Alert } from 'reactstrap';

const AlertError = ({ error }) => {
	let returnVal = null;
	if (error) {
		returnVal = (
			<Alert color="danger">
				{error}
			</Alert>
		);
	}
	return returnVal;
};

export default AlertError;
