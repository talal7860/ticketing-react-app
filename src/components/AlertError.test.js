import React from 'react';
import { shallow } from 'enzyme';
import AlertError from './AlertError';
import { Alert } from 'reactstrap';

describe('AlertError', () => {
  it('should render an Alert when an error is provided to it', () => {
    const wrapper = shallow(<AlertError error="error" />);

    expect(wrapper).toMatchSnapshot();
  });

});

