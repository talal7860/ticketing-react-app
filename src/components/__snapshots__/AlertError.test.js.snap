// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertError should render an Alert when an error is provided to it 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlertError
    error="error"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "error",
      "closeAriaLabel": "Close",
      "color": "danger",
      "isOpen": true,
      "tag": "div",
      "transition": Object {
        "appear": true,
        "baseClass": "fade",
        "baseClassActive": "show",
        "enter": true,
        "exit": true,
        "in": true,
        "mountOnEnter": false,
        "onEnter": [Function],
        "onEntered": [Function],
        "onEntering": [Function],
        "onExit": [Function],
        "onExited": [Function],
        "onExiting": [Function],
        "tag": "div",
        "timeout": 150,
        "unmountOnExit": true,
      },
    },
    "ref": null,
    "rendered": "error",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "error",
        "closeAriaLabel": "Close",
        "color": "danger",
        "isOpen": true,
        "tag": "div",
        "transition": Object {
          "appear": true,
          "baseClass": "fade",
          "baseClassActive": "show",
          "enter": true,
          "exit": true,
          "in": true,
          "mountOnEnter": false,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "tag": "div",
          "timeout": 150,
          "unmountOnExit": true,
        },
      },
      "ref": null,
      "rendered": "error",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
